pyfar/plot/two_d.py

```diff
@@ -5,6 +5,98 @@
 
 
+def time2d(signal, dB=False, log_prefix=20, log_reference=1, unit=None,
+           points=None, orientation="vertical",
+           cmap=mpl.cm.get_cmap(name='magma'), colorbar=True, ax=None,
+           style='light', **kwargs):
+    """
+    Surface plot of the time signal.
+
+    Plots ``signal.time`` and passes keyword arguments (`kwargs`) to
+    ``matplotlib.pyplot.pcolormesh()``.
+
+    Parameters
+    ----------
+
+    Parameters
+    ----------
+    signal : Signal, TimeData
+        The input data to be plotted.
+    dB : bool
+        Indicate if the data should be plotted in dB in which case
+        ``log_prefix * np.log10(signal.time / log_reference)`` is used. The
+        default is ``False``.
+    log_prefix : integer, float
+        Prefix for calculating the logarithmic time data. The default is
+        ``20``.
+    log_reference : integer
+        Reference for calculating the logarithmic time data. The default is
+        ``1``.
+    unit : str, None
+        Unit of the time axis. Can be ``s``, ``ms``, ``mus``, or ``samples``.
+        The default is ``None``, which sets the unit to ``s`` (seconds), ``ms``
+        (milli seconds), or ``mus`` (micro seconds) depending on the data.
+    points: array like, optional
+        Points at which the channels of `signal` were sampled (e.g. azimuth
+        angles or x values). `points` must be monotonously increasing/
+        decreasing and have as many entries as `signal` has channels. The
+        default is ``'None'`` which labels the channels in `signal` from
+        0 to N.
+    orientation: string, optional
+        ``'vertical'``
+            The channels of `signal` will be plotted as as vertical lines.
+        ``'horizontal'``
+            The channels of `signal` will be plotted as horizontal lines.
+
+        The default is ``'vertical'``
+    ax : matplotlib.pyplot.axes
+        Axes to plot on. The default is ``None``, which uses the current axis
+        or creates a new figure if none exists.
+    style : str
+        ``light`` or ``dark`` to use the pyfar plot styles or a plot style from
+        ``matplotlib.style.available``. The default is ``light``.
+
+
+    Returns
+    -------
+    ax : matplotlib.pyplot.axes
+        Axes or array of axes containing the plot.
+
+    Examples
+    --------
+
+    Plot a multichannel signal with information about the points where it was
+    samples
+
+    .. plot::
+        >>> import pyfar as pf
+        >>> import numpy as np
+        >>> # generate the signal
+        >>> angles = np.arange(0, 180)
+        >>> delays = np.round(100 * np.sin(angles / 180 * np.pi)).astype(int)
+        >>> signal = pf.signals.impulse(128, delays)
+        >>> # plot the signal
+        >>> pf.plot.time2d(signal, points=angles)
+    """
+
+    with context(style):
+        ax, qm, cb = _two_d._time2d(
+            signal.flatten(), dB, log_prefix, log_reference, unit,
+            points, orientation, cmap, colorbar, ax, **kwargs)
```

Comment:
Signals of cshape higher than 1D are flattened. This is not compatible with points haveing the same shape as cshape:
```
import pyfar as pf
import numpy as np
# generate the signal
angles = np.ones((2, 180)) * np.arange(0, 180)
delays = np.round(100 * np.sin(angles / 180 * np.pi)).astype(int)
signal = pf.signals.impulse(128, delays)
# plot the signal
pf.plot.time2d(signal, points=angles, orientation='horizontal')
```
Suggestion:
```suggestion
            signal.flatten(), dB, log_prefix, log_reference, unit,
            np.asarray(points).flatten(), orientation, cmap, colorbar, ax, **kwargs)
```
What about describing this behavior explicitly in the docs?
Also, this is a bit different for spectrogram, where you cycle through the channels. Maybe it would make sense to have the cycling as a second option?

Reply:
Tue, I did not think about that. In this case I would suggest to raise an error if a signal has a multi-dimensional `cshape`. Flattening the signal would make less sense in the cases that I am thinking of.

Reply:
Agree :)